host-controller/src/main/java/org/jboss/as/host/controller/operations/StartServersHandler.java

```diff
@@ -95,27 +96,47 @@ public void execute(OperationContext context, ModelNode operation) throws Operat
                 final Resource resource =  context.readResource(PathAddress.EMPTY_ADDRESS);
                 final ModelNode hostModel = Resource.Tools.readModel(resource);
                 if(hostModel.hasDefined(SERVER_CONFIG)) {
+
                     final ModelNode servers = hostModel.get(SERVER_CONFIG).clone();
+                    final boolean restartRequired = runningModeControl.getRestartMode() == RestartMode.SERVERS;
+                    final boolean blockUntilStopped = hostControllerEnvironment.isRestart() || !runningModeControl.isReloadByHandler() || (runningModeControl.isReloadByHandler() && runningModeControl.isBlockUntilStopped());
```

Comment:
Is there any difference between these two cases?

hostControllerEnvironment.isRestart() || !runningModeControl.isReloadByHandler()

In other words if this op is invoked during a boot where hostControllerEnvironment.isRestart(), won't !runningModeControl.isReloadByHandler() also be true? AIUI isReloadByHandler() is true for any reload and is false for any non-reload boot.

Also, HostControllerEnvironment.isRestart is final, so for the life of an HC process started that way isRestart() will never return 'false' and therefore blockUntilStopped will always be 'true'. Which doesn't seem right.

Reply:
I didn't realize that HostControllerEnvironment.isRestart was final. The intention was to wait for servers until stopped always if we were restarting the HC process, but being a final variable, it will affect always independent of isReloadByHandler or isBlockUntilStopped.

@bstansberry Changed to use only a single condition: runningModeControl.isBlockUntilStopped(), it only gets false if the user has used a reload(blocking=false), otherwise, we wait.